homeassistant/components/knx/switch.py

```diff
@@ -34,32 +36,33 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up switch(es) for KNX platform."""
-    xknx: XKNX = hass.data[DOMAIN].xknx
-    config: list[ConfigType] = hass.data[DATA_KNX_CONFIG][Platform.SWITCH]
+    knx_module: KNXModule = hass.data[DOMAIN]
 
-    async_add_entities(KNXSwitch(xknx, entity_config) for entity_config in config)
+    yaml_config: list[ConfigType] | None
+    if yaml_config := hass.data[DATA_KNX_CONFIG].get(Platform.SWITCH):
+        async_add_entities(
+            KnxYamlSwitch(knx_module.xknx, entity_config)
+            for entity_config in yaml_config
+        )
+    ui_config: dict[str, ConfigType] | None
+    if ui_config := knx_module.config_store.data["entities"].get(Platform.SWITCH):
+        async_add_entities(
+            KnxUiSwitch(knx_module, unique_id, config)
+            for unique_id, config in ui_config.items()
+        )
```

Comment:
How often will both be true? If it will be for most of the time, rather group the entities and call `async_add_entities` once

Reply:
I guess this is user dependent. Some will prefer yaml, others UI. But I guess a lot will at least try to configure one or some UI entities while leaving existing on yaml... so both would be true for them.